.ci/Jenkinsfile

```diff
@@ -375,6 +378,11 @@ def withCloudTestEnv(Closure body) {
     [var: "AWS_ACCESS_KEY_ID", password: aws.access_key],
     [var: "AWS_SECRET_ACCESS_KEY", password: aws.secret_key],
   ])
+  // GCP
+  withCredentials([file(credentialsId: env.JOB_GCS_EXT_CREDENTIALS, variable: 'FILE_CREDENTIAL')]) {
+    maskedVars.add([var: 'GOOGLE_CREDENTIALS', password: readFile(file: env.FILE_CREDENTIAL)]);
+    maskedVars.add([var: 'GCP_PROJECT_ID', password: env.GCP_PROJECT_ID])
+  }
```

Comment:
You don't need this section but a shared library step to wrap the command that requires the GCP connection:

- https://github.com/elastic/apm-pipeline-library/blob/main/vars/withGCPEnv.txt

This step allows to use the credentials stored in vault, this is much more dynamic. You can see an example in:
- https://github.com/elastic/golang-crossbuild/blob/f89836691dfbc6784d096150ac1acb5458ed63c2/Jenkinsfile#L162-L170

Reply:
I can experiment with `withGCPEnv`, but all we need is to sniff the GOOGLE_CREDENTIALS and pass it to the TF container (only this ENV, maybe also GCP_PROJECT_ID). We don't need full initialization of `withGCPEnv`. Once we pull the credentials, we can disable the `withGCPEnv`'s context.

Let me know what you think.

Reply:
Just in case: I adapted the PR to use `withGCP`